src/shapirowilk.jl

```diff
@@ -0,0 +1,248 @@
+    println(io, indent, rpad("W-statistic:", l), t.W)
+end
+
+function pvalue(t::ShapiroWilkTest)
+    n = length(t.SWc)
+    W = t.W
+
+    if iszero(censored_ratio(t))
+        if n == 3 # exact by Shapiro&Wilk 1965
+            # equivalent to 6/π * (asin(sqrt(W)) - asin(sqrt(3/4)))
+            return 1 - 6acos(sqrt(W)) / π
+        elseif n ≤ 11 # Royston 1992
+            γ = __RS92_G(n)
+            if γ ≤ log1p(-W)
+                return zero(W)
+            end
+            w = -log(γ - log1p(-W))
+            μ = __RS92_C3(n)
+            σ = exp(__RS92_C4(n))
+        elseif 12 ≤ n # Royston 1992
+            w = log1p(-W)
+            μ = __RS92_C5(log(n))
+            σ = exp(__RS92_C6(log(n)))
+        end
+        return ccdf(Normal(μ, σ), w)
+    else
+        throw("censored samples not implemented yet")
+        # to implement censored samples follow Royston 1993 Section 3.3
+    end
+end
+
+"""
+    ShapiroWilkTest(X::AbstractArray{<:Real}, SWc::SWCoeffs=SWCoeffs(length(X)); kwargs...)
+Perform a Shapiro-Wilk test of normality on `X`.
+
+This julia implementation is based the method of Royston (1992).
+The calculation of the p-value is exact for `N = 3`, and for ranges
+`4 ≤ N ≤ 11` and `12 ≤ N ≤ 5000` (Royston 1992) two separate approximations
+for p-values are used.
+
+Implements: [`pvalue`](@ref)
+
+# Notes (Royston 1993)
+* While the (approximated) W-statistic will be accurate for large sample size
+  (`N > 2000`), returned p-values may not be reliable.
+* Censoring too much data (`(N - N1) / N > 0.8`, where `N1` is (upper)
+  uncensored data length), or when the sample size is small (`N < 20`) may
+  produce unreliable p-values.
+
+# Implementation notes
+* The current implementation DOES NOT implement p-values for censored data.
+* If multiple Shapiro-Wilk tests are to be performed on samples of same
+  cardinality it is beneficial to pass `SWc` for re-use.
+* For maximal performance sorted `X` should be passed and indicated with
+  `sample_sorted=true` keyword argument.
+
+# References
+Shapiro, S. S., & Wilk, M. B. (1965). An Analysis of Variance Test for Normality
+(Complete Samples). *Biometrika*, 52, 591–611.
+[doi:10.1093/BIOMET/52.3-4.591](https://doi.org/10.1093/BIOMET/52.3-4.591).
+
+Royston, P. (1992). Approximating the Shapiro-Wilk W-test for non-normality.
+*Statistics and Computing*, 2(3), 117–119.
+[doi:10.1007/BF01891203](https://doi.org/10.1007/BF01891203)
+
+Royston, P. (1993). A Toolkit for Testing for Non-Normality in Complete and
+Censored Samples. Journal of the Royal Statistical Society Series D
+(The Statistician), 42(1), 37–43.
+[doi:10.2307/2348109](https://doi.org/10.2307/2348109)
+
+Royston, P. (1995). Remark AS R94: A Remark on Algorithm AS 181: The W-test for
+Normality. *Journal of the Royal Statistical Society Series C
+(Applied Statistics)*, 44(4), 547–551.
+[doi:10.2307/2986146](https://doi.org/10.2307/2986146).
+"""
+function ShapiroWilkTest(
```

Comment:
This isn't the indentation style we use in general in the package. Better put the first argument on the first line, and align others with it.

Reply:
Is there a `.JuliaFormatter.toml` with the style?

Reply:
The style used in JuliaStats is the same as that used in Julia Base, which I think is more or less the same as YAS. But I can't swear it will work... @ararslan must know.

Reply:
ok, HT doesn't use any consistent style, see #299; I just tried to be more or less consistent here